### DISCUSSION TITLE
Inquiry About Compatibility with libcrypt API

### DISCUSSION THREAD

**Discussion by samirMhn** [openssl/openssl#25607] [Q&A]
Dear OpenSSL Team,

I hope this message finds you well.

I am reaching out to inquire whether OpenSSL provides an API that is compatible with the libcrypt API from glibc. We are currently using unix_pam, which is linked to libcrypt, and I would like to understand if OpenSSL has an API that can work seamlessly with libcrypt.

Thank you for your assistance. I look forward to your response.

Best regards,

**Comment by t8m** [comment]
No, OpenSSL does not provide such API. There is an `openssl passwd` application that can produce password hashes compatible with libcrypt however this is not a library implementation of the crypt API.

**Reply by samirMhn** [reply]
Thank you for your response.

Even if OpenSSL doesn’t provide a library compatible with libcrypt, can we still use libcrypto to achieve the same functionality as the "openssl passwd" command?

I’m asking this because I’m using unix_pam for authentication, and pam_unix relies on libcrypt. I wanted to see if I could use OpenSSL (libcrypto or another component) to develop a pam_openssl module based on OpenSSL, which would perform the same tasks as pam_unix

**Reply by t8m** [reply]
It depends on which password hash types you want to implement. For those implemented in `openssl passwd` you can get it from the openssl app source code and of course that will call OpenSSL for crypto operations. There are additional password hash types which use algorithms currently not implemented by OpenSSL though.

**Reply by samirMhn** [reply]
Yes, it's essentially SHA-256, SHA-512, MD5 (the ones I need are all available on openssl)... I searched the internet to see if there are any implementations of pam_unix based on OpenSSL, but I couldn't find any
